kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/ReplicationControllerOperationsImpl.java

```diff
@@ -151,13 +151,17 @@ private void waitUntilRCIsScaled() {
 
     final Runnable rcPoller = new Runnable() {
       public void run() {
-        ReplicationController rc = getMandatory();
-        atomicRC.set(rc);
-        if (Objects.equals(rc.getSpec().getReplicas(), rc.getStatus().getReplicas())) {
-          countDownLatch.countDown();
-        } else {
-          LOG.debug("Only {}/{} replicas scheduled for ReplicationController: {} in namespace: {} seconds so waiting...",
+        try {
+          ReplicationController rc = getMandatory();
+          atomicRC.set(rc);
+          if (Objects.equals(rc.getSpec().getReplicas(), rc.getStatus().getReplicas())) {
```

Comment:
There's nothing defending against that here but it shouldn't be possible via remote API I don't think.